### ISSUE TITLE
Read only attribute causing NoMethodError (undefined method `map' for nil:NilClass)

### ISSUE THREAD

**Issue by mikaelprag** [mailjet/mailjet-gem#236]
Updating a record that does not have the array `read_only_attributes` fails with the error message:
`NoMethodError (undefined method 'map' for nil:NilClass)`

Code to reproduce:
```
target = Mailjet::Contactdata.find(1)
target.update_attributes(
  data: [
    {
      'Name' => 'first_name',
      'Value' => 'John'
    }
  ]
)
```

I think the code to blame is the one below and that a check needs to happen to see if this array exists or not. I'll give a PR a shot.

https://github.com/mailjet/mailjet-gem/blob/acbd7221149dd1682f6db4a91d45cd7d5ecbf7aa/lib/mailjet/resource.rb#L321

**Comment by maxencehenneron** [comment]
I have the same issue.

**Comment by Retttro** [comment]
Thank you for your contribution. Your suggested PR https://github.com/mailjet/mailjet-gem/pull/237 is merged and was released in the 1.7.2 version. Closing the issue.
